### PR TITLE
REF: Make more things properties; .md is inspectable before run

### PR DESCRIPTION
**Pull request by danielballan**
Tests pass, but I want to discuss the approach before merging.

### CODE REVIEW

**Comment by danielballan** [comment]
tl;dr We were using `_pre_scan` to set a lot of things that could have just been properties. Now that they are, one can inspect `plan.md` at any time.

This would also have solved #236. I still think the changes in #236 were good, but this should be done as well. Does this seem like a step in the right direction, @klauer?

**Comment by danielballan** [comment]
Test pass.

File: bluesky/scans.py [hunk 1]
```diff
@@ -782,21 +806,22 @@ def args(self):
...
             # Special case first pass because their is no
             # mutliplicative identity for cyclers.
-            if self.cycler is None:
-                self.cycler = c
+            if result is None:
+                result = c
             else:
-                self.cycler += c
-        yield from super()._pre_scan()
+                result += c
```
Comment (tacaswell) [inline bluesky/scans.py]: Probably better to make a list of cyclers and then run the list through a reduce.

File: bluesky/scans.py [hunk 2]
```diff
@@ -812,9 +837,12 @@ class InnerProductAbsScan(_InnerProductScanBase):
     motor1, start1, stop1, ..., motorN, startN, stopN : list
         motors can be any 'setable' object (motor, temp controller, etc.)
     """
-    def _pre_scan(self):
-        self._offsets = defaultdict(lambda: 0)
-        yield from super()._pre_scan()
+    _derived_fields = ['motors', 'extents']
+
+    @property
+    def init_pos(self):
```
Comment (tacaswell) [inline bluesky/scans.py]: make private(ish)

File: bluesky/scans.py [hunk 3]
```diff
@@ -866,9 +909,11 @@ class OuterProductAbsScan(_OuterProductScanBase):
         is a boolean indicating whether to following snake-like, winding
         trajectory or a simple left-to-right trajectory.
     """
-    def _pre_scan(self):
-        self._offsets = defaultdict(lambda: 0)
-        yield from super()._pre_scan()
+    _derived_fields = ['motors', 'shape', 'num', 'extents', 'snaking']
+    @property
+    def init_pos(self):
```
Comment (tacaswell) [inline bluesky/scans.py]: private(ish)

**Comment by danielballan** [comment]
I did more cleanup. This code is less confusing to read now, I hope. Ready for one last review.

**Comment by danielballan** [comment]
@tacaswell has reviewed this a couple times and I have @klauer's humorously curmudgeonly blessing.

**Pull request merged**